src/Datadog.Trace.ClrProfiler.Managed/Emit/ModuleLookup.cs

```diff
@@ -0,0 +1,46 @@
+using System;
+using System.Collections.Concurrent;
+using System.Reflection;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.ClrProfiler.Emit
+{
+    internal static class ModuleLookup
+    {
+        private static readonly ILog Log = LogProvider.GetLogger(typeof(ModuleLookup));
+
+        private static ConcurrentDictionary<Guid, Module> _modules = new ConcurrentDictionary<Guid, Module>();
+
+        public static Module Get(Guid moduleVersionId)
+        {
+            if (!_modules.ContainsKey(moduleVersionId))
+            {
+                PopulateModules();
+            }
+
+            if (_modules.ContainsKey(moduleVersionId))
```

Comment:
Using two separate operations on the dictionary is less efficient and can lead to race conditions. For example, `_modules.ContainsKey()` can return `true`, then another thread can remove that key, before `_modules[moduleVersionId]` is called, which will result in an exception.

Use `_modules.TryGetValue()` instead:
```csharp
if(_modules.TryGetValue(moduleVersionId, out Module module)) ? module : null;
```

Reply:
Should likely use a semaphore to prevent multiple PopulateCalls from running at once as well.